sys/net/gnrc/link_layer/lwmac/rx_state_machine.c

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_lwmac
+ * @{
+ *
+ * @file
+ * @brief       Implementation of RX state machine of LWMAC protocol
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#include "net/gnrc.h"
+#include "net/gnrc/lwmac/lwmac.h"
+#include "net/gnrc/mac/internal.h"
+#include "net/gnrc/lwmac/timeout.h"
+#include "include/rx_state_machine.h"
+#include "include/lwmac_internal.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define LOG_LEVEL LOG_WARNING
+#include "log.h"
+
+/**
+ * @brief   Flag to track if the receiver has got a broadcast packet
+ */
+#define GNRC_LWMAC_RX_FOUND_BROADCAST         (0x01U)
+
+/**
+ * @brief   Flag to track if the receiver has got a WR packet
+ */
+#define GNRC_LWMAC_RX_FOUND_WR                (0x02U)
+
+/**
+ * @brief   Flag to track if the receiver has got a data packet
+ */
+#define GNRC_LWMAC_RX_FOUND_DATA              (0x04U)
+
+#undef LOG_ERROR
+#undef LOG_WARNING
+#undef LOG_INFO
+#undef LOG_DEBUG
+
+#define LOG_ERROR(...) LOG(LOG_ERROR, "ERROR: [lwmac-rx] " __VA_ARGS__)
+#define LOG_WARNING(...) LOG(LOG_WARNING, "WARNING: [lwmac-rx] " __VA_ARGS__)
+#define LOG_INFO(...) LOG(LOG_INFO, "[lwmac-rx] " __VA_ARGS__)
+#define LOG_DEBUG(...) LOG(LOG_DEBUG, "[lwmac-rx] " __VA_ARGS__)
+
+/* Break out of switch and mark the need for rescheduling */
+#define GOTO_RX_STATE(rx_state, do_resched) gnrc_netdev->rx.state = rx_state; \
+                                            reschedule = do_resched; \
+                                            break
```

Comment:
For the two first instructions you can use an inline function. The break I would write out in any case, other wise the code is harder to read.